### ISSUE TITLE
salt/cloud/ec2.py encoding problems.

### ISSUE THREAD

**Issue by Modulus** [saltstack/salt#35088]
### Setup

A node with a tag containing the letter "Ø" crashes the command salt-cloud -m maps/mymap.map
We have a similar setup in development, but in production that letter is here, since it is in the name of the account admin. 
### Steps to Reproduce Issue

Simply create a tag on any node in your setup containing the letter "Æ, Ø or Å" (copy from this text)

The error you se when you run the command with "-l debug" is as follows:

```
[DEBUG   ] Failed to execute 'ec2.list_nodes()' while querying for running nodes: 'ascii' codec can't encode character u'\xd8' in position 165422: ordinal not in range(128)
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/salt/cloud/__init__.py", line 2364, in run_parallel_map_providers_query
    cloud.clouds[data['fun']]()
  File "/usr/lib/python2.7/dist-packages/salt/cloud/clouds/ec2.py", line 3425, in list_nodes
    nodes = list_nodes_full(get_location())
  File "/usr/lib/python2.7/dist-packages/salt/cloud/clouds/ec2.py", line 3275, in list_nodes_full
    return _list_nodes_full(location)
  File "/usr/lib/python2.7/dist-packages/salt/cloud/clouds/ec2.py", line 3361, in _list_nodes_full
    sigver='4')
  File "/usr/lib/python2.7/dist-packages/salt/utils/aws.py", line 463, in query
    result.text
UnicodeEncodeError: 'ascii' codec can't encode character u'\xd8' in position 165422: ordinal not in range(128)
The following virtual machines are set to be created:
```

The unicode character for xd8 is "Ø". But why is this using ascii encoding and not unicode? Would this not fail if there are tags in German, Icelandic, Russian, Chinese etc..???
### Versions Report

```
Salt Version:
           Salt: 2016.3.1

Dependency Versions:
           cffi: Not Installed
       cherrypy: 3.2.2
       dateutil: 2.5.3
          gitdb: 0.5.4
      gitpython: 0.3.2 RC1
          ioflo: Not Installed
         Jinja2: 2.7.2
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: 0.21.1
           Mako: 0.9.1
   msgpack-pure: Not Installed
 msgpack-python: 0.3.0
   mysql-python: 1.2.3
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: Not Installed
         Python: 2.7.6 (default, Jun 22 2015, 17:58:13)
   python-gnupg: Not Installed
         PyYAML: 3.10
          PyZMQ: 14.0.1
           RAET: Not Installed
          smmap: 0.8.2
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.0.4

System Versions:
           dist: Ubuntu 14.04 trusty
        machine: x86_64
        release: 3.13.0-74-generic
         system: Linux
        version: Ubuntu 14.04 trusty
```

**Comment by Ch3LL** [comment]
I am seeing the same behavior when running the following command: `sudo salt-cloud -a set_tags ch3ll_test tag1='Øtag' -ldebug`

ping @techhat do you know if there is a better way to handle these characters in tags?

**Comment by rsuarezsoto** [comment]
I think I'm being affected by this as well, but in a different way. Some of our AWS resources have Unicode characters in descriptions, and that seems to break salt-cloud. Example:

```
# salt-cloud -p trusty_t2_micro trustycloudtest
[WARNING ] There's no master defined on the 'trustycloudtest' VM settings.
[DEBUG   ] Generating minion keys for 'trustycloudtest'
[DEBUG   ] LazyLoaded cloud.fire_event
[DEBUG   ] MasterEvent PUB socket URI: /var/run/salt/master/master/master_event_pub.ipc
[DEBUG   ] MasterEvent PULL socket URI: /var/run/salt/master/master/master_event_pull.ipc
[DEBUG   ] Initializing new IPCClient for path: /var/run/salt/master/master/master_event_pull.ipc
[INFO    ] Creating Cloud VM trustycloudtest in eu-west-1
[DEBUG   ] Create network interface: {'DeviceIndex': 0, 'allocate_new_eip': True}
[DEBUG   ] Using AWS endpoint: ec2.eu-west-1.amazonaws.com
[DEBUG   ] AWS Request: https://ec2.eu-west-1.amazonaws.com/?Action=DescribeSubnets&Version=2014-10-01
[INFO    ] Starting new HTTPS connection (1): ec2.eu-west-1.amazonaws.com
[DEBUG   ] Setting read timeout to None
[DEBUG   ] "GET /?Action=DescribeSubnets&Version=2014-10-01 HTTP/1.1" 200 None
[DEBUG   ] AWS Response Status Code: 200
[ERROR   ] There was a profile error: 'ascii' codec can't encode character u'\xf3' in position 9432: ordinal not in range(128)
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/salt/cloud/cli.py", line 284, in run
    self.config.get('names')
  File "/usr/lib/python2.7/dist-packages/salt/cloud/__init__.py", line 1451, in run_profile
    ret[name] = self.create(vm_)
  File "/usr/lib/python2.7/dist-packages/salt/cloud/__init__.py", line 1281, in create
    output = self.clouds[func](vm_)
  File "/usr/lib/python2.7/dist-packages/salt/cloud/clouds/ec2.py", line 2446, in create
    data, vm_ = request_instance(vm_, location)
  File "/usr/lib/python2.7/dist-packages/salt/cloud/clouds/ec2.py", line 1735, in request_instance
    _new_eni = _create_eni_if_necessary(interface, vm_)
  File "/usr/lib/python2.7/dist-packages/salt/cloud/clouds/ec2.py", line 1249, in _create_eni_if_necessary
    sigver='4')
  File "/usr/lib/python2.7/dist-packages/salt/utils/aws.py", line 463, in query
    result.text
UnicodeEncodeError: 'ascii' codec can't encode character u'\xf3' in position 9432: ordinal not in range(128)
```

u'\xf3' is the character "ó", which indeed we use in one of our subnet descriptions.

I'm running this command in a vagrant box, Ubuntu 14.04. Versions report for salt-cloud:
```
# salt-cloud --versions-report
Salt Version:
            Salt: 2016.3.3
 
Dependency Versions:
 Apache Libcloud: 0.15.1
            cffi: Not Installed
        cherrypy: Not Installed
        dateutil: 1.5
           gitdb: Not Installed
       gitpython: Not Installed
           ioflo: Not Installed
          Jinja2: 2.7.2
         libgit2: Not Installed
         libnacl: Not Installed
        M2Crypto: Not Installed
            Mako: 0.9.1
    msgpack-pure: Not Installed
  msgpack-python: 0.4.6
    mysql-python: 1.2.3
       pycparser: Not Installed
        pycrypto: 2.6.1
          pygit2: Not Installed
          Python: 2.7.6 (default, Jun 22 2015, 17:58:13)
    python-gnupg: Not Installed
          PyYAML: 3.10
           PyZMQ: Not Installed
            RAET: Not Installed
           smmap: Not Installed
         timelib: Not Installed
         Tornado: 4.2.1
             ZMQ: Not Installed
 
System Versions:
            dist: Ubuntu 14.04 trusty
         machine: x86_64
         release: 3.13.0-96-generic
          system: Linux
         version: Ubuntu 14.04 trusty
```

Thanks in advance!

**Comment by damon-atkins** [comment]
Link to #37692 `Change Salt to default to Unicode for STR in Develop`

**Comment by rallytime** [comment]
@Modulus I think this should be fixed by #37912, and also #41514.

**Comment by Modulus** [comment]
Great thanks :)
